internal/querynodev2/services.go

```diff
@@ -253,8 +246,7 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 
 	node.manager.Collection.PutOrRef(req.GetCollectionID(), req.GetSchema(),
 		node.composeIndexMeta(req.GetIndexInfoList(), req.Schema), req.GetLoadMeta())
-	collection := node.manager.Collection.Get(req.GetCollectionID())
-	collection.SetMetricType(req.GetLoadMeta().GetMetricType())
```

Comment:
Why remove this logic?
Previously, this logic was implemented to support the ability to search without specifying the metric_type.
The metric_type is already present in the loaded index, so users do not need to specify it